filecoin-proofs/src/api/update.rs

```diff
@@ -652,3 +664,264 @@ pub fn verify_empty_sector_update_proof<Tree: 'static + MerkleTreeTrait<Hasher =
+/// requires these inputs.
+///
+/// This method allows them to be retrieved when needed, rather than storing them for
+/// some amount of time.
+///
+/// # Arguments
+///
+/// * `porep_config` - this sector's porep config that contains the number of bytes in the sector.
+/// * `comm_r_old` - a commitment to a sector's previous replica.
+/// * `comm_r_new` - a commitment to a sector's current replica.
+/// * `comm_d_new` - a commitment to a sector's current data.
+pub fn get_sector_update_inputs<Tree: 'static + MerkleTreeTrait<Hasher = TreeRHasher>>(
+    porep_config: &PoRepConfig,
+    comm_r_old: Commitment,
+    comm_r_new: Commitment,
+    comm_d_new: Commitment,
+) -> Result<Vec<Vec<Fr>>> {
+    trace!("get_sector_update_inputs:start");
+
+    // Note that there's nothing inherently incorrect about zero
+    // commitments, but given that this check exists during the
+    // sealing process and may have historically been hit, this is
+    // considered a consistency check
+    ensure!(
+        comm_r_old != [0; 32],
+        "Invalid all zero commitment (comm_r_old)"
+    );
+    ensure!(
+        comm_r_new != [0; 32],
+        "Invalid all zero commitment (comm_r_new)"
+    );
+    ensure!(
+        comm_d_new != [0; 32],
+        "Invalid all zero commitment (comm_d_new)"
+    );
+
+    let comm_r_old_safe = <TreeRHasher as Hasher>::Domain::try_from_bytes(&comm_r_old)?;
+    let comm_r_new_safe = <TreeRHasher as Hasher>::Domain::try_from_bytes(&comm_r_new)?;
+
+    let comm_d_new_safe = DefaultPieceDomain::try_from_bytes(&comm_d_new)?;
+
+    let config = SectorUpdateConfig::from_porep_config(porep_config);
+    let partitions = usize::from(config.update_partitions);
+
+    let public_inputs: storage_proofs_update::PublicInputs = PublicInputs {
+        k: 0,
+        comm_r_old: comm_r_old_safe,
+        comm_d_new: comm_d_new_safe,
+        comm_r_new: comm_r_new_safe,
+        h: config.h,
+    };
+    let setup_params_compound = compound_proof::SetupParams {
+        vanilla_params: SetupParams {
+            sector_bytes: u64::from(config.sector_size),
+        },
+        partitions: Some(partitions),
+        priority: false,
+    };
+    let pub_params_compound = EmptySectorUpdateCompound::<Tree>::setup(&setup_params_compound)?;
+
+    // These are returned for aggregated proof verification.
+    let inputs: Vec<_> = (0..partitions)
+        .into_par_iter()
+        .map(|k| {
+            EmptySectorUpdateCompound::<Tree>::generate_public_inputs(
+                &public_inputs,
+                &pub_params_compound.vanilla_params,
+                Some(k),
+            )
+        })
+        .collect::<Result<_>>()?;
+
+    trace!("get_sector_update_inputs:finish");
+
+    Ok(inputs)
+}
+
+// Hash all of the commitments into an ordered digest for the aggregate proof method.
+fn get_hashed_commitments(sector_update_inputs: &[SectorUpdateProofInputs]) -> [u8; 32] {
```

Comment:
Just flagging that the public input `h` (SnapDeals security parameter) may or may not need to be included here. Atm it's a constant (and thus wouldn't need to be hashed into the SnarkPack transcript), but it's technically not a constant because it can be incremented if SnapDeals' current security level is ever deemed insufficient.

If `h` should be included in the SnarkPack transcript (the SnarkPack transcript should include a digest of the non-constant Groth16 public inputs), I think that this function could be modified to include `h` without adding too much Sha256 hashing overhead.
```rust
let mut hasher = Sha256::new();
for input in sector_update_inputs.iter() {
    // This would require `SectorUpdateInputs` to also have an `h` field. 
    hasher.update(&[input.h as u8]);
    hasher.update(input.comm_r_old);
    hasher.update(input.comm_r_new);
    hasher.update(input.comm_d_new);
}
hasher.finalize()
```
Because `h` will be a constant for the foreseeable future, I don't think that it needs to be included; I'm mainly just flagging it for the SuperSnaps audit.